src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ObjectInitializerCompletionProviderTests.cs

```diff
@@ -847,6 +847,26 @@ public static T Create<T>()
             await VerifyItemExistsAsync(markup, "B");
         }
 
+        [WorkItem(24612, "https://github.com/dotnet/roslyn/issues/24612")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task ObjectInitializerOfGenericTypeСonstraint2()
+        {
+            var markup = @"
+internal class Example
+{
+    public static T Create<T>()
+        where T : new()
```

Comment:
can we also do "where T : struct" and "where T : System.Delegate" and "where T : unmanaged"?

Reply:
@CyrusNajmabadi, I'll add a test with ```where T : struct``` constraint.
Regarding other cases:
1. ```where T : System.Delegate```: "Constraint cannot be special class 'Delegate'"
2. ```where T : unmanaged```: I'm not sure what do you mean. It is not valid syntax.

Reply:
I thought it was.  See: https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.3/blittable.md

@OmarTawfik can you advise?  Has this been added to the language?

Reply:
Yes. You've to use C# 7.3 for the new syntax to work :)
Please LMK/mention me if you see any issues.